tidb/src/main/java/io/tidb/bigdata/tidb/codec/TiDBEncodeHelper.java

```diff
@@ -390,23 +397,12 @@ public List<BytePairWrapper> generateKeyValuesToDeleteByRow(Row row) {
     // get deletion row
     if (handleCol != null || isCommonHandle) {
       Handle handle = extractHandle(row);
-      Optional<Row> optRow = getOldRowWithHandleColOrIsCommonHandle(handle, snapshot);
-      optRow.ifPresent(oldRow -> deletion.add(new Pair<>(oldRow, handle)));
+      getOldRowWithHandleColOrIsCommonHandle(handle, snapshot).ifPresent(deletion::add);
```

Comment:
WithClusteredIndex

Reply:
it will also work in TiDB 4.0 when pkisHandle. it is not only for ClusteredIndex